datadog/data_source_datadog_service_level_objectives.go

```diff
@@ -0,0 +1,108 @@
+		Schema: map[string]*schema.Schema{
+			"ids": {
+				Description: "An array of SLO IDs to limit the search.",
+				Type:        schema.TypeList,
+				Optional:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"name_query": {
+				Description: "Filter results based on SLO names.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"tags_query": {
+				Description: "Filter results based on a single SLO tag.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"metrics_query": {
+				Description: "Filter results based on SLO numerator and denominator.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+
+			// Computed values
+			"slos": {
+				Description: "List of SLOs",
+				Type:        schema.TypeList,
+				Computed:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Description: "Id of Datadog service level objective",
+							Type:        schema.TypeString,
+							Computed:    true,
+						},
+						"name": {
+							Description: "Name of Datadog service level objective",
+							Type:        schema.TypeString,
+							Computed:    true,
+						},
+						"type": {
+							Description: "The type of the service level objective. The mapping from these types to the types found in the Datadog Web UI can be found in the Datadog API [documentation page](https://docs.datadoghq.com/api/v1/service-level-objectives/#create-a-slo-object). Available options to choose from are: `metric` and `monitor`.",
+							Type:        schema.TypeString,
+							Computed:    true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceDatadogServiceLevelObjectivesRead(d *schema.ResourceData, meta interface{}) error {
+	providerConf := meta.(*ProviderConfiguration)
+	datadogClientV1 := providerConf.DatadogClientV1
+	authV1 := providerConf.AuthV1
+
+	req := datadogClientV1.ServiceLevelObjectivesApi.ListSLOs(authV1)
+	if v, ok := d.GetOk("ids"); ok {
+		req = req.Ids(strings.Join(expandStringList(v.([]interface{})), ","))
+	}
+	if v, ok := d.GetOk("name_query"); ok {
+		req = req.Query(v.(string))
+	}
+	if v, ok := d.GetOk("tags_query"); ok {
+		req = req.TagsQuery(v.(string))
+	}
+	if v, ok := d.GetOk("metrics_query"); ok {
+		req = req.MetricsQuery(v.(string))
+	}
+
+	slosResp, _, err := req.Execute()
+	if err != nil {
+		return utils.TranslateClientError(err, "error querying service level objectives")
+	}
+	if len(slosResp.GetData()) == 0 {
+		return fmt.Errorf("your query returned no result, please try a less specific search criteria")
+	}
+
+	slos := make([]map[string]interface{}, 0, len(slosResp.GetData()))
+	for _, slo := range slosResp.GetData() {
+		slos = append(slos, map[string]interface{}{
+			"id":   slo.GetId(),
+			"name": slo.GetName(),
+			"type": slo.GetType(),
+		})
+	}
+	if err := d.Set("slos", slos); err != nil {
+		return err
+	}
+	d.SetId("datadog-service-level-objectives")
```

Comment:
I think since the data-source allows filtering, and isn't a "global"/"unique" data-source that returns all SLOs, we'll need to give it a unique ID. Maybe something similar to - https://github.com/DataDog/terraform-provider-datadog/blob/master/datadog/data_source_datadog_security_monitoring_rules.go#L140